### PR TITLE
Enable the message right-click context menu in the browser

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: enhancement

<hr>

This PR enables showing the right-click context menu in the browser unless we are clicking selected text or an `a` tag

<hr>

https://user-images.githubusercontent.com/25768714/163592335-96e575d3-cbe5-49ad-a67a-62a8250e5f9c.mp4

---
Here's what your changelog entry will look like:

## ✨ Features
 * Enable the message right-click context menu in the browser ([\#8336](https://github.com/matrix-org/matrix-react-sdk/pull/8336)). Contributed by @SimonBrandner.

Preview: https://pr8336--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/EventTile.tsx [hunk 1]
```diff
@@ -947,13 +947,26 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
...
     private showContextMenu(ev: React.MouseEvent, showPermalink?: boolean): void {
+        // Return if message right-click context menu isn't enabled
         if (!SettingsStore.getValue("feature_message_right_click_context_menu")) return;
+
+        // Return if we're in a browser and click either an a tag or we have
+        // selected text, as in those cases we want to use the native browser
+        // menu
+        const clickTarget = ev.target as HTMLElement;
+        if (
+            !PlatformPeg.get().allowOverridingNativeContextMenus() &&
```
Comment (turt2live) [inline src/components/views/rooms/EventTile.tsx:958]:
```suggestion
            !PlatformPeg.get().allowOverridingNativeContextMenus() ||
```

to maintain the check that was removed later on?

Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx:958]: We don't want to maintain that check - we want to allow the menu even in the browser; we just want to make sure we don't show the custom menu in the browser when the other conditions are met

Comment (turt2live) [reply src/components/views/rooms/EventTile.tsx:958]: but surely if the user has highlighted text or right clicked a link then we'd want to show the OS menu on desktop too?

Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx:958]: [The agreement seemed to be to replace the OS menu on the desktop with the custom one](https://user-images.githubusercontent.com/60790/159077851-ee96736b-d568-49c1-8241-8192b034ef8e.png), I am not sure if there have been any developments, I assume this picture is from an internal repo, to which I don't have access...

Comment (turt2live) [reply src/components/views/rooms/EventTile.tsx:958]: hmm, alright. The image is from an internal repo (where there's not much more information than the screenshot).

File: src/components/views/context_menus/MessageContextMenu.tsx [hunk 2]
```diff
@@ -58,6 +58,10 @@ export function canCancel(status: EventStatus): boolean {
     return status === EventStatus.QUEUED || status === EventStatus.NOT_SENT || status === EventStatus.ENCRYPTING;
 }
 
+export function getSelectedText(): string {
```
Comment (turt2live) [inline src/components/views/context_menus/MessageContextMenu.tsx:61]: this should be moved to a utility somewhere

**Review by turt2live — COMMENTED** [review]
otherwise lgtm - this doesn't need design review as a labs flag at the moment

**Review by turt2live — APPROVED** [review]

**Pull request merged**